test-validator/src/lib.rs

```diff
@@ -348,7 +389,14 @@ impl TestValidatorGenesis {
             addresses,
             rpc_client,
             skip_missing,
-            |_address, account| Ok(AccountSharedData::from(account)),
+            |address, account| {
+                let account_shared_data = AccountSharedData::from(account);
+                // ignore the error
+                Ok(
+                    try_transform_program_data(address, account_shared_data.clone())
+                        .unwrap_or(account_shared_data),
+                )
```

Comment:
Is this clone here avoidable? If `try_transform_program_data` took a mutable reference, couldn't you do this here?

```rust
let mut account_shared_data = AccountSharedData::from(account);
try_transform_program_data(&mut account_shared_data).ok(); // Error still ignored.
account_shared_data
```

And then in the other function:

```rust
let mut account_shared_data = AccountSharedData::from(account);
try_transform_program_data(&mut account_shared_data)?;
account_shared_data
```

Reply:
Ah yeah that's a great idea, done!